core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java

```diff
@@ -1533,6 +1557,47 @@ public void testDelete()
                         new Identifier("b")))));
     }
 
+    @Test
+    public void testMerge()
+    {
+        assertStatement("" +
+                        "MERGE INTO inventory AS i\n" +
+                        "  USING changes AS c\n" +
+                        "  ON i.part = c.part\n" +
+                        "WHEN MATCHED AND c.action = 'mod'\n" +
+                        "  THEN UPDATE SET\n" +
+                        "    qty = qty + c.qty\n" +
+                        "  , ts = CURRENT_TIMESTAMP\n" +
+                        "WHEN MATCHED AND c.action = 'del'\n" +
+                        "  THEN DELETE\n" +
+                        "WHEN NOT MATCHED AND c.action = 'new'\n" +
+                        "  THEN INSERT (part, qty) VALUES (c.part, c.qty)",
+                new Merge(
+                        new NodeLocation(1, 1),
+                        table(QualifiedName.of("inventory")),
+                        Optional.of(new Identifier("i")),
+                        aliased(table(QualifiedName.of("changes")), "c"),
+                        equal(nameReference("i", "part"), nameReference("c", "part")),
+                        ImmutableList.of(
+                                new MergeUpdate(
+                                        new NodeLocation(1, 1),
+                                        Optional.of(equal(nameReference("c", "action"), new StringLiteral("mod"))),
+                                        ImmutableList.of(
+                                                new MergeUpdate.Assignment("qty", new ArithmeticBinaryExpression(
+                                                        ArithmeticBinaryExpression.Operator.ADD,
+                                                        nameReference("qty"),
+                                                        nameReference("c", "qty"))),
+                                                new MergeUpdate.Assignment("ts", new CurrentTime(CurrentTime.Function.TIMESTAMP)))),
+                                new MergeDelete(
+                                        new NodeLocation(1, 1),
```

Comment:
I added the missing constructors that omit the `NodeLocations` for the `Merge*` classes, and which allowed me to remove the `NodeLocation` arguments in this test.

Reply:
Thanks for adding those additional constructors.

However, the previous version of your test exposed an issue: you can provide a `NodeLocation`, and it will be silently ignored, even if it's wrong. That is because the assertion is based on `Node`'s `equals()` method, which doesn't take the location into account.
I think we should disallow passing the `NodeLocation` in `TestSqlParser`. I'll file an issue.